### PR TITLE
🚏 Auto-generate hooks from typechain

### PR DESCRIPTION
**Pull request by rzadp**
## Next steps

Add yargs for command line parameters instead of env vars only

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,5 @@ packages/example/gen/types
 dist
 packages/.pnpm-debug.log
 packages/example-next/.next
+packages/core/src/gen
+dist-node
```
Comment (rzadp) [inline .gitignore:10]:
```suggestion
```

File: packages/example/package.json [hunk 2]
```diff
@@ -58,20 +63,26 @@
...
+    "ethereum-waffle": "4.0.0-alpha.0",
     "html-webpack-plugin": "^4.5.0",
+    "jsdom": "^20.0.0",
+    "jsdom-global": "^3.0.2",
     "mocha": "^8.2.1",
     "playwright": "^1.20.2",
     "prettier": "^2.0.5",
     "react-refresh": "^0.10.0",
     "rimraf": "^3.0.2",
+    "solc": "^0.8.12",
```
[the remark this answers is not in the document]
Comment (jakvbs) [reply packages/example/package.json:81]: There is an error on older version of typechain

File: packages/core/src/hooks/useToken.test.tsx [hunk 3]
```diff
@@ -23,12 +23,13 @@ describe('useToken', async () => {
...
     expect(result.error).to.be.undefined
-    expect(result.current).to.deep.equal({
+    const res = {
       name: 'MOCKToken',
       symbol: 'MOCK',
       decimals: 18,
       totalSupply: MOCK_TOKEN_INITIAL_BALANCE,
-    })
+    }
+    expect(JSON.parse(JSON.stringify(result.current))).to.deep.equal(JSON.parse(JSON.stringify(res)))
```
Comment (rzadp) [inline packages/core/src/hooks/useToken.test.tsx:32]: We can check it again separately in between tasks.

File: packages/core/generate-hooks/generate.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    let output = commonImports + imports({typesDir, abisDir, outDir, contractName})
+    console.log(`Processing ${contractName}`)
+    const factory = factories[factoryName]
+    const Interface = factory.createInterface()
+    
+    const abi = factory.abi
+    Object.keys(Interface.functions).forEach((fn) => {
+      const functionName = fn.split('(')[0]
+      const fnABI = abi.find((a: any) => a.name === functionName)
+      if (fnABI?.stateMutability === 'view') {
```
Comment (yivlad) [inline packages/core/generate-hooks/generate.ts:30]: `['view', 'pure'].includes(fnABI?.stateMutability)`?

File: packages/core/tsconfig.json [hunk 5]
```diff
@@ -1,6 +1,10 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
+    "types": [
+      "mocha",
+      "node"
+    ],
     "outDir": "dist",
     "module": "commonjs",
```
Comment (yivlad) [inline packages/core/tsconfig.json:9]: I also had to add `generate-hooks` to `includes` so my VS Code could do typechecking, not sure what's the problem

**Review by yivlad — APPROVED** [review]

File: packages/core/tsconfig.json (hunk 5 above)
Comment (jakvbs) [reply packages/core/tsconfig.json:9]: Introduced

**Pull request merged**